Source/Engine/Utilities/Extensions.cs

```diff
@@ -404,5 +238,160 @@ public static void Shuffle<T>(this Random random, IList<T> collection)
                 collection[n] = value;
             }
         }
+
+        /// <summary>
+        /// Generates a random <see cref="bool"/>.
+        /// </summary>
+        /// <param name="random">An instance of <see cref="Random"/>.</param>
+        /// <param name="weight">Normalized value that determines the chance to return true.</param>
+        /// <returns>A <see cref="bool"/> thats either true or false.</returns>
+        public static bool NextBool(this Random random, float weight = 0.5f) => random.NextDouble() < weight;
+
+        /// <summary>
+        /// Generates a random <see cref="byte"/> value between min and max.
+        /// </summary>
+        /// <param name="random">An instance of <see cref="Random"/>.</param>
+        /// <param name="min">The minimum value.</param>
+        /// <param name="max">The maximum value.</param>
+        /// <returns>A random <see cref="byte"/> between min and max.</returns>
+        public static byte NextByte(this Random random, byte min = 0, byte max = byte.MaxValue)
```

Comment:
This is a sneaky off-by-one bug, since the maximum is exclusive. Meaning `random.Next(0, 255);` won't ever return 255.

Reply:
Fixed.